packages/contracts-bedrock/test/fixtures/L2GenesisFixtures.sol

```diff
@@ -0,0 +1,288 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Script } from "forge-std/Script.sol";
+import { console2 as console } from "forge-std/console2.sol";
```

Comment:
I am confused how we are supposed to maintain this file. I don't think its a good idea to hardcode the L2 contract bytecode into a library and assert based on exactness. 

We have a few options on how to do this type of assertion. I think a good option would be to wrap individual setter functions in the L2 genesis script with the cheatcodes introduced here: https://github.com/foundry-rs/forge-std/pull/481
Then we can get a dump of the exact state being set and make assertions on a per contract basis

Reply:
I'm understanding correctly, I believe one problem we'll run into is situations like the recent ecotone changes to L1Block:
- L1Block has been updated to include ecotone specific code
- The L1Block predeploy currently does not have that code, and therefore its compiled code would not (and should not) match the hardcoded bytecode
- This test starts failing because the bytecode from compiling L1Block doesn't match what's hardcoded
- The generated genesis file now uses WIP contract code

So do we have a broader problem of: how do we ensure that the genesis JSON is using the most recent released version of predeploys, without including WIP code?

Reply:
> how do we ensure that the genesis JSON is using the most recent released version of predeploys, without including WIP code?

Just throwing this idea out there, maybe we can install the latest released version of the contracts as Foundry deps and pull those into the genesis generation script? This would be the local version of the predeploys can be whatever, but we're only setting the genesis alloc to latest released version

Reply:
> how do we ensure that the genesis JSON is using the most recent released version of predeploys, without including WIP code?

We can come up with a solution for this but for now we just need to be mindful of it. I believe the solution would be to define bedrock as ecotone style upgrade txs and then apply ecotone upgrade txs then optionally apply the local source code on top. This gives us releases of L2 contracts but then we can also test against the source code changes

h/t @protolambda for this idea

Reply:
>> how do we ensure that the genesis JSON is using the most recent released version of predeploys, without including WIP code?

> We can come up with a solution for this but for now we just need to be mindful of it.

It would probably good to document this somewhere, but I'm not sure what a good place would be. It should be the same place we document how to generate the L2 genesis and the config options - I imagine this exists somewhere?

Reply:
Updated, removed hardcoded deployed bytecode and storage slots